src/Mod/Draft/DraftGui.py

```diff
@@ -2402,29 +2404,42 @@ def createObject(self):
         except Exception as e:
             FreeCAD.Console.PrintError("Draft_ShapeString: error delaying commit\n")
 
-    def platWinDialog(self, OnOff):
-        tDialog = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/Dialog")
-        if platform.system() == 'Windows':
-            if OnOff:
-                return tDialog.SetBool("DontUseNativeDialog", True)
-            else:
-                return tDialog.SetBool("DontUseNativeDialog", False)
+    def platWinDialog(self, Flag):
+        ParamGroup = FreeCAD.ParamGet("User parameter:BaseApp/Preferences/Dialog")
+        if Flag == "Overwrite":
+            GroupContent = ParamGroup.GetContents()
+
+            Found = False
+            for ParamSet in GroupContent:
+              if ParamSet[1] == "DontUseNativeFontDialog":
+                Found = True
+                break
+
+            if Found == False:
+                ParamGroup.SetBool("DontUseNativeFontDialog", True)     #initialize nonexisting one
+
+            param = ParamGroup.GetBool("DontUseNativeFontDialog")
+            ShapeStringTaskPanel.oldValueBuffer = ParamGroup.GetBool("DontUseNativeDialog")
```

Comment:
This can be `self.oldValueBuffer`. You don't have to mention the entire class name again, `ShapeStringTaskPanel`.